src/librustdoc/doctest.rs

```diff
@@ -349,6 +349,12 @@ fn run_test(
     compiler.env("UNSTABLE_RUSTDOC_TEST_PATH", path);
     compiler.env("UNSTABLE_RUSTDOC_TEST_LINE", format!("{}", line as isize - line_offset as isize));
     compiler.arg("-o").arg(&output_file);
+
+    // Enable passed lints
+    for (lint_name, level) in &rustdoc_options.lint_opts {
+        compiler.arg(level.to_cmd_flag()).arg(lint_name);
+    }
```

Comment:
I am unsure if this approach is correct. Please double check this block with https://github.com/rust-lang/rust/blob/91178d27b4c38f5f2768ea86c731ecaca09fa4ac/src/bootstrap/builder.rs#L1736

Reply:
yes, this is not correct: we should only enable it for crates in the rust-lang/rust workspace, not all rust crates anywhere in the ecosystem.

Please also use `-Zcrate-attr`, not attributes in each crate, to ensure the lint is automatically enabled when someone adds a new rustc_ crate. See https://github.com/rust-lang/rust/blob/3b1c8a94a4e8a6ba8bc7b39cc3580db9e5b72295/src/bootstrap/compile.rs#L393 for an example.